waku/waku_archive/driver/postgres_driver/migrations.nim

```diff
@@ -0,0 +1,95 @@
+{.push raises: [].}
+
+import
+  std/[tables, strutils, os],
+  stew/results,
+  chronicles,
+  chronos
+import
+  ../../../common/databases/common,
+  ../../../../migrations/message_store_postgres/pg_migration_manager,
+  ../postgres_driver
+
+logScope:
+  topics = "waku archive migration"
+
+const SchemaVersion* = 1 # increase this when there is an update in the database schema
+
+proc breakIntoStatements*(script: string): seq[string] =
+  ## Given a full migration script, that can potentially contain a list
+  ## of SQL statements, this proc splits it into the contained isolated statements
+  ## that should be executed one after the other.
+  var statements = newSeq[string]()
+
+  let lines = script.split('\n')
+
+  var simpleStmt: string
+  var plSqlStatement: string
+  var insidePlSqlScript = false
+  for line in lines:
+    if line.strip().len == 0:
+      continue
+
+    if insidePlSqlScript:
+      if line.contains("END $$"):
+        ## End of the Pl/SQL script
+        plSqlStatement &= line
+        statements.add(plSqlStatement)
+        plSqlStatement = ""
+        insidePlSqlScript = false
+        continue
+
+      else:
+        plSqlStatement &= line & "\n"
+
+    if line.contains("DO $$"):
+      ## Beginning of the Pl/SQL script
+      insidePlSqlScript = true
+      plSqlStatement &= line & "\n"
+
+    if not insidePlSqlScript:
+      if line.contains(';'):
+        ## End of simple statement
+        simpleStmt &= line
+        statements.add(simpleStmt)
+        simpleStmt = ""
+      else:
+        simpleStmt &= line & "\n"
+
+  return statements
+
+proc migrate*(driver: PostgresDriver,
+              targetVersion = SchemaVersion):
+              Future[DatabaseResult[void]] {.async.} =
```

Comment:
Same comment, as it is the source place: 
Is it necessary to have postgres driver's migrate async while the prototype and sqlLite version is sync?

Reply:
> Same comment, as it is the source place: Is it necessary to have postgres driver's migrate async while the prototype and sqlLite version is sync?

Ah yes, every interaction with _Postgres_ should be async.
We could decorate _SQLite_ procs/methods as `async` but it is sync underneath.